packages/react-devtools-shared/src/devtools/views/Components/Components.js

```diff
@@ -17,35 +24,189 @@ import portaledContent from '../portaledContent';
+                      <SelectedElement />
+                    </Suspense>
+                  </NativeStyleContextController>
+                </div>
+                <ModalDialog />
+                <SettingsModal />
+              </Fragment>
+            )}
+          </ComponentResizer>
         </InspectedElementContextController>
       </OwnersListContextController>
     </SettingsModalContextController>
   );
 }
 
+type HorizontalResizeDirection = 'HORIZONTAL';
+type VerticalResizeDirection = 'VERTICAL';
+
+const RESIZE_DIRECTIONS: {|
+  HORIZONTAL: HorizontalResizeDirection,
+  VERTICAL: VerticalResizeDirection,
+|} = {
+  HORIZONTAL: 'HORIZONTAL',
+  VERTICAL: 'VERTICAL',
+};
+
+const LOCAL_STORAGE_RESIZE_ELEMENT_PERCENTAGE_HORIZONTAL_KEY = `React::DevTools::resizedElementPercentage::${RESIZE_DIRECTIONS.HORIZONTAL}`;
+const LOCAL_STORAGE_RESIZE_ELEMENT_PERCENTAGE_VERTICAL_KEY = `React::DevTools::resizedElementPercentage::${RESIZE_DIRECTIONS.VERTICAL}`;
+
+function ComponentResizer({children}): {|children: Function|} {
+  const [isResizing, setIsResizing] = useState<boolean>(false);
+  const [
+    horizontalPercentage,
+    setHorizontalPercentage,
+  ] = useLocalStorage<number>(
+    LOCAL_STORAGE_RESIZE_ELEMENT_PERCENTAGE_HORIZONTAL_KEY,
+    65,
+  );
+  const [verticalPercentage, setVerticalPercentage] = useLocalStorage<number>(
+    LOCAL_STORAGE_RESIZE_ELEMENT_PERCENTAGE_VERTICAL_KEY,
+    50,
+  );
+  const updateLocalStorageTimeoutId = useRef<number>(null);
+  const componentsWrapperRef = useRef<HTMLDivElement>(null);
+  const resizeElementRef = useRef<HTMLElement>(null);
+  const [resizeElementStyles, setResizeElementStyles] = useState<Object>({});
+  const getResizeDirection: Function = useCallback(() => {
+    if (componentsWrapperRef.current === null) {
+      return RESIZE_DIRECTIONS.HORIZONTAL;
+    }
+    const {width} = componentsWrapperRef.current.getBoundingClientRect();
+
+    return width > 600
+      ? RESIZE_DIRECTIONS.HORIZONTAL
+      : RESIZE_DIRECTIONS.VERTICAL;
+  }, [componentsWrapperRef]);
+
+  const onResizeStart = useCallback(() => {
+    setIsResizing(true);
+  }, [setIsResizing]);
+
+  const onResizeEnd = useCallback(() => {
+    setIsResizing(false);
+  }, [setIsResizing]);
+
+  const onResize = useCallback(
+    e => {
+      if (
+        !isResizing ||
+        componentsWrapperRef.current === null ||
+        resizeElementRef.current === null
+      ) {
+        return;
+      }
+
+      e.preventDefault();
+
+      const {
+        height,
+        width,
+        left,
+        top,
+      } = componentsWrapperRef.current.getBoundingClientRect();
+      const resizeDirection = getResizeDirection();
+      const currentMousePosition: number =
+        resizeDirection === RESIZE_DIRECTIONS.HORIZONTAL
+          ? e.clientX - left
+          : e.clientY - top;
+      const boundary: {|
+        min: number,
+        max: number,
+      |} = {
+        min: 40,
```

Comment:
Maybe move `40` to a constant?
```suggestion
        min: BOUNDARY_PADDING,
```

Reply:
Added 🥂